### ISSUE TITLE
Update Darknet component for GPU portability

### ISSUE THREAD

**Issue by jrobble** [openmpf/openmpf#530]
Once #527 is complete, determine what changes need to be made to the build process (cmake files, etc.) to enable maximum portability across a variety of GPU architectures. This involves determining which nvcc options to use for CUDA support.

Compile the Darknet component libary with GPU support on a development VM and then run it on various HPC clusters with different real GPU hardware to ensure portability.

**Comment by brosenberg42** [comment]
I used the CMake FindCuda module (https://cmake.org/cmake/help/v3.6/module/FindCUDA.html) to configure the build. I think you will basically have to tinker with the `CUDA_NVCC_FLAGS` variable. In my initial build script I tried to just use the default options as much as possible. I set `CUDA_NVCC_FLAGS` to `set(CUDA_NVCC_FLAGS -DGPU -DOPENCV --compiler-options -fPIC )` From what I understand `--compiler-options` are options that get passed down to the underlying C compiler, so I think the `-gencode` options should appear before `--compiler-options`.
